thrust/async/scan.h

```diff
@@ -0,0 +1,355 @@
+/*
```

Comment:
We should have separate headers for each of the algorithms.

Reply:
I'm going to hold off on this -- we should think about more granular headers in general in thrust. Since the `thrust::` scan algorithms are all in a single public header, I'd rather the async public headers match those conventions.